x-pack/test/security_solution_ftr/services/timeline/index.ts

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Response } from 'superagent';
+import { EndpointError } from '@kbn/security-solution-plugin/common/endpoint/errors';
+import { TIMELINE_DRAFT_URL, TIMELINE_URL } from '@kbn/security-solution-plugin/common/constants';
+import { TimelineResponse } from '@kbn/security-solution-plugin/common/types';
+import { TimelineInput } from '@kbn/security-solution-plugin/common/search_strategy';
+import moment from 'moment';
+import { FtrService } from '../../../functional/ftr_provider_context';
+
+export class TimelineTestService extends FtrService {
+  private readonly supertest = this.ctx.getService('supertest');
+  private readonly log = this.ctx.getService('log');
+
+  /**
+   * Returns an error handler for `supertest` request that will dump out more useful information
+   * when things fail.
+   *
+   * @param ignoredStatusCodes
+   * @private
+   *
+   * @example
+   *
+   * await this.supertest
+   *    .post('/some/api')
+   *    .set('kbn-xsrf', 'true')
+   *    .send(somePayLoad)
+   *    .then(this.getHttpResponseFailureHandler([409]));
+   */
+  private getHttpResponseFailureHandler(
+    ignoredStatusCodes: number[] = []
+  ): (res: Response) => Promise<Response> {
+    return async (res) => {
+      if (!res.ok && !ignoredStatusCodes.includes(res.status)) {
+        throw new EndpointError(JSON.stringify(res.error, null, 2));
+      }
+
+      return res;
+    };
+  }
+
+  /**
+   * Creates a new timeline.
+   *
+   * Note: Although the timeline is created, when displayed on the UI, no events are retrieved
```

Comment:
Perhaps I missed something as to why this does not "just work". I did several diff's from timelines created via the UI with the code here and could not find anything that would make it work without having to trigger an update from the UI after it is displayed.

If anyone from the Threat Hunting team knows of a trick, let me know _(I did all of this just by reverse eng. what the UI appeared to be be doing)_